packages/dds/tree/src/simple-tree/treeNodeKernel.ts

```diff
@@ -31,9 +34,24 @@ export class TreeNodeKernel implements Listenable<TreeChangeEvents> {
 	public constructor(public readonly node: TreeNode) {}
 
 	public hydrate(anchorNode: AnchorNode): void {
-		const offChildrenChanged = anchorNode.on("childrenChangedAfterBatch", () => {
-			this.#events.emit("nodeChanged");
-		});
+		const offChildrenChanged = anchorNode.on(
+			"childrenChangedAfterBatch",
+			({ changedFields }) => {
+				const flexNode = anchorNode.slots.get(flexTreeSlot);
+				assert(flexNode !== undefined, "Flex node does not exist");
+				const nodeSchema = getSimpleNodeSchema(flexNode.schema);
+				const changedProperties = isObjectNodeSchema(nodeSchema)
+					? new Set(
+							[...changedFields].map(
```

Comment:
nit: `Array.from(changedFields, (field) => ...)` is a more direct way to produce an array that is mapped from an iterable (only makes one array instead of two).

Reply:
Didn't know that overload existed, thanks!